front/src/modules/timesStops/hooks/useTimesStopsTableData.ts

```diff
@@ -0,0 +1,248 @@
+import { useEffect, useState, useMemo } from 'react';
+
+import { keyBy } from 'lodash';
+import { useTranslation } from 'react-i18next';
+import { useSelector } from 'react-redux';
+
+import usePathOps from 'applications/operationalStudies/hooks/usePathOps';
+import { useScenarioContext } from 'applications/operationalStudies/hooks/useScenarioContext';
+import type { PathPropertiesFormatted } from 'applications/operationalStudies/types';
+import { matchOpRefAndOp } from 'applications/operationalStudies/utils';
+import type { SimulationResponseSuccess, TrackSection } from 'common/api/osrdEditoastApi';
+import { matchPathStepAndOp } from 'modules/pathfinding/utils';
+import { interpolateValue } from 'modules/simulationResult/helpers/utils';
+import type { SimulationSummary } from 'modules/timetableItem/types';
+import type { Train } from 'reducers/osrdconf/types';
+import { getDisplayOnlyPathSteps } from 'reducers/simulationResults/selectors';
+import { Duration } from 'utils/duration';
+
+import { getTrackReferenceLabel, getOperationalPointName } from '../helpers/utils';
+import type { ScheduleEntry, TimesStopsRowNew } from '../types';
+
+type BuildTableRowParams = {
+  id: string;
+  name?: string;
+  ch?: string;
+  trackName?: string;
+  startDatetime: Date;
+  schedule?: ScheduleEntry;
+  computedArrivalMs?: number;
+};
+
+const buildTableRow = ({
+  id,
+  name,
+  ch,
+  trackName,
+  startDatetime,
+  schedule,
+  computedArrivalMs,
+}: BuildTableRowParams): TimesStopsRowNew => {
+  // schedule.arrival is ISO 8601 duration
+  const requestedArrivalDuration = schedule?.arrival ? Duration.parse(schedule.arrival) : null;
+  const requestedArrival = requestedArrivalDuration
+    ? new Date(startDatetime.getTime() + requestedArrivalDuration.ms)
+    : null;
+
+  // computedArrivalMs is offset in ms from startDatetime
+  const computedArrival =
+    computedArrivalMs !== undefined ? new Date(startDatetime.getTime() + computedArrivalMs) : null;
+
+  // schedule.stop_for is ISO 8601 duration
+  const stopDuration = schedule?.stop_for ? Duration.parse(schedule.stop_for) : null;
+
+  // requestedDeparture = requestedArrival + stopDuration
+  const requestedDeparture =
+    requestedArrival && stopDuration
```

Comment:
Even if `stopDuration` is 0, you should take it into account. Same remark l.62
```suggestion
    requestedArrival && stopDuration !== null
```